api/v1beta1/vaultauth_types.go

```diff
@@ -93,11 +93,23 @@ type VaultAuthConfigAWS struct {
 // VaultAuthSpec defines the desired state of VaultAuth
 type VaultAuthSpec struct {
 	// VaultConnectionRef of the corresponding VaultConnection CustomResource.
+	// The VaultConnectionRef may be prefixed with a namespace, eg: `namespaceA/connectionB`.
+	// If no namespace is included the Operator will default to the namespace of the VaultAuth CR.
 	// If no value is specified the Operator will default to the `default` VaultConnection,
-	// configured in its own Kubernetes namespace.
+	// configured in its own Kubernetes namespace unless prefixed.
 	VaultConnectionRef string `json:"vaultConnectionRef,omitempty"`
 	// Namespace to auth to in Vault
 	Namespace string `json:"namespace,omitempty"`
+	// AllowedNamespaces Kubernetes Namespaces which are allow-listed for use with this AuthMethod.
+	// This field allows administrators to customize which Kubernetes namespaces are authorized to
+	// act with this AuthMethod. While Vault will still enforce its own rules, this has the added
+	// configurability of restricting which VaultAuthMethods can be used by which namespaces.
+	// Accepted values:
+	// []{"*"} - wildcard, all namespaces.
+	// []{"a", "b"} - list of namespaces.
+	// []{} - empty list, no namespaces.
+	// nil - all namespaces, default behavior.
```

Comment:
Does this mean they can assign the value `nil`. Maybe we could say `unset or empty then the VaultAuth can only be referenced from a Vault*Secret's namespace` ?  I think that nil slice and empty slice are the same.

I wonder if we could make it so that VSO will allow the namespace if it is in `AllowedNamespaces` or `== *` ?

Reply:
Refactored this per our conversation.
Fixed in a52da2d